src/main/java/TaskList.java

```diff
@@ -0,0 +1,188 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    private String filePath = "./data/tasks.txt";
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    public Task getTask(int taskIndex) {
+        return tasks.get(taskIndex);
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    public void deleteTask(int taskIndex) throws BuddyException {
+        if (tasks.size() == 0 ) {
+            System.out.println("There are no tasks in your list:");
+        } else if (taskIndex >= 0 && taskIndex < tasks.size()) {
+            Task deletedTask = tasks.remove(taskIndex);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(deletedTask.toString());
+            if (tasks.size() == 1) {
```

Comment:
You don't need to specifically identify if the size is equal to 1 and instead can combine the two cases into one since in both cases they print out the same statement